src/lib/marlin_plonk_bindings/stubs/src/arkworks/bigint_256.rs

```diff
@@ -0,0 +1,239 @@
+        self.0.clone().into()
+    }
+}
+
+impl TryFrom<BigUint> for CamlBigInteger256 {
+    type Error = String;
+
+    fn try_from(x: BigUint) -> Result<Self, Self::Error> {
+        Ok(Self(BigInteger256::try_from(x)?))
+    }
+}
+
+impl TryFrom<&BigUint> for CamlBigInteger256 {
+    type Error = String;
+
+    fn try_from(x: &BigUint) -> Result<Self, Self::Error> {
+        Ok(Self(BigInteger256::try_from(x.clone())?))
+    }
+}
+
+impl ToString for CamlBigInteger256 {
+    fn to_string(&self) -> String {
+        Into::<BigUint>::into(self).to_string()
+    }
+}
+
+//
+// OCaml stuff
+//
+
+#[ocaml::func]
+pub fn caml_bigint_256_of_numeral(
+    s: &[u8],
+    _len: u16,
+    base: u16,
+) -> Result<CamlBigInteger256, ocaml::Error> {
+    match BigUint::parse_bytes(s, base as u32) {
+        Some(data) => CamlBigInteger256::try_from(data)
+            .map_err(|_| ocaml::Error::Message("caml_bigint_256_of_numeral")),
+        None => Err(ocaml::Error::Message("caml_bigint_256_of_numeral")),
+    }
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_of_decimal_string(s: &[u8]) -> Result<CamlBigInteger256, ocaml::Error> {
+    match BigUint::parse_bytes(s, 10) {
+        Some(data) => CamlBigInteger256::try_from(data)
+            .map_err(|_| ocaml::Error::Message("caml_bigint_256_of_decimal_string")),
+        None => Err(ocaml::Error::Message("caml_bigint_256_of_decimal_string")),
+    }
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_num_limbs() -> ocaml::Int {
+    return BIGINT256_NUM_LIMBS.try_into().unwrap();
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_bytes_per_limb() -> ocaml::Int {
+    return BIGINT256_LIMB_BYTES.try_into().unwrap();
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_div(x: CamlBigInteger256, y: CamlBigInteger256) -> CamlBigInteger256 {
+    let x: BigUint = x.into();
+    let y: BigUint = y.into();
+    let res: BigUint = x / y;
+    let inner: BigInteger256 = res.try_into().expect("BigUint division has a bug");
+    inner.into()
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_compare(
+    x: ocaml::Pointer<CamlBigInteger256>,
+    y: ocaml::Pointer<CamlBigInteger256>,
+) -> ocaml::Int {
+    match x.as_ref().cmp(y.as_ref()) {
+        Less => -1,
+        Equal => 0,
+        Greater => 1,
+    }
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_print(x: CamlBigInteger256) {
+    println!("{}", x.to_string());
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_to_string(x: CamlBigInteger256) -> String {
+    x.to_string()
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_test_bit(
+    x: ocaml::Pointer<CamlBigInteger256>,
+    i: ocaml::Int,
+) -> Result<bool, ocaml::Error> {
+    match i.try_into() {
+        Ok(i) => Ok(x.as_ref().get_bit(i)),
+        Err(_) => Err(ocaml::Error::invalid_argument("caml_bigint_256_test_bit")
+            .err()
+            .unwrap()),
+    }
+}
+
+#[ocaml::func]
+pub fn caml_bigint_256_to_bytes(x: ocaml::Pointer<CamlBigInteger256>) -> ocaml::Value {
+    let len = std::mem::size_of::<BigInteger256>();
```

Comment:
as you pointed out this is code that was already here :D 
but I agree with your comments, I think this should be refactored to avoid the unsafe. Maybe in a separate PR?